psycopg_pool/psycopg_pool/pool_async.py

```diff
@@ -648,13 +647,13 @@ async def worker(cls, q: AQueue[MaintenanceTask]) -> None:
     async def _connect(self, timeout: float | None = None) -> ACT:
         """Return a new connection configured for the pool."""
         self._stats[self._CONNECTIONS_NUM] += 1
-        kwargs = self.kwargs
+        conninfo, kwargs = await self._resolve_conninfo(), await self._resolve_kwargs()
```

Comment:
No need to pack and unpack a tuple hee.
```suggestion
        conninfo = await self._resolve_conninfo()
        Kwargs = await self._resolve_kwargs()
```

Reply:
done.